### PR TITLE
Upgrade Spek to v2.0.8

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

**Comment by 3flex** [comment]
I'm guessing the failure is due to a behaviour change in Spek meaning `described` nested in `describe` fails where it didn't before.

Changing `describe` to `context` here should fix it: https://github.com/arturbosch/detekt/blob/1c93b1e9def6054fb17c26c7655ff31ad9bcda7a/detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/FileBasedFindingsReportSpec.kt#L16

**Comment by schalkms** [comment]
@3flex I'm on it. That's only a part of the problem.

File: detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/FileBasedFindingsReportSpec.kt [hunk 1]
```diff
@@ -35,7 +35,7 @@ class FileBasedFindingsReportSpec : Spek({
                     Pair("EmptySmells", emptyList())
                 )
             }
-            val output = subject.render(detektion)?.trimEnd()?.decolorized()
+            val output by memoized { subject.render(detektion)?.trimEnd()?.decolorized() }
```
Comment (3flex) [inline detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/FileBasedFindingsReportSpec.kt]: I think it's better to put this line in the `it()` context... the Spek docs say not to use memoized for holding the result of an action https://www.spekframework.org/core-concepts/#scope-values

So instead:
```kotlin
            it("has the reference content") {
                val output = subject.render(detektion)?.trimEnd()?.decolorized()
                assertThat(output).isEqualTo(expectedContent)
```
Will that work?

Comment (schalkms) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/FileBasedFindingsReportSpec.kt]: With the updated commit with `lint`/`compileAndLint` for each and every test case, which is not really what we want.

Comment (schalkms) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/FileBasedFindingsReportSpec.kt]: With the updated commit we `lint`/`compileAndLint` the code for each and every test case, which is not really what we want.
BeforeEachGroup also doesn't work.

Comment (3flex) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/FileBasedFindingsReportSpec.kt]: Another option is putting all assertions into a single test case for the tests that assert multiple things for each test file.

I think what you have here now though is ok. This problem will go away when test cases are moved out of the external files and into individual test cases.

**Pull request merged**